config/cmake/cacheinit.cmake

```diff
@@ -43,6 +43,8 @@ set (HDF5_MINGW_STATIC_GCC_LIBS ON CACHE BOOL "Statically link libgcc/libstdc++"
 set (HDF5_ALLOW_EXTERNAL_SUPPORT "TGZ" CACHE STRING "Allow External Library Building (NO GIT TGZ)" FORCE)
 set_property (CACHE HDF5_ALLOW_EXTERNAL_SUPPORT PROPERTY STRINGS NO GIT TGZ)
 
+set (HDF5_SET_DEFAULT_RPATH ON "Set the default RPATH for the common installation" FORCE)
```

Comment:
Presumably this is just going to be a problem for `ccmake` builds since the cacheinit file isn't necessarily going to be used.

Reply:
Why? ccmake could use the presets file and I will have a a followup PR that adds the dependent_option form to all  options that are behind another option.

Reply:
I'd say for the average user building from source there really is no reason to use the presets, cacheinit, etc. files. The default configuration you get when just running `ccmake` on a build directory should be the happy path.

Reply:
The defaults should build a limited but workable solution. cacheinit.cmake is mostly for our testing as the settings match our expected/typical build. presets are useful for use cases.